Growth and integrations handbook page (#1446)
sourcegraph/handbook: content/product-engineering/engineering/cloud/growth-and-integrations/index.md

        <div style="text-align: center; font-size: 12px;">Software Engineer</div>
        <div style="text-align: center; font-size: 12px;">(starting 01/10)</div>
      </div>
    </div>

<div class="col" style="flex: 1;">
      <div>
        <div>
          <img src="https://storage.googleapis.com/sourcegraph-assets/handbook/growth-and-integrations/rob-rhyne.png" alt="Picture of Rob Rhyne" style="background: transparent; width:128px;"/>
        </div>
        <div style="text-align: center;">Rob Rhyne</div>
        <div style="text-align: center; font-size: 12px;">Temporary product manager and product designer, growth</div>
      </div>
    </div>
    <div class="col" style="flex: 1;">
      <div>
        <div>
          <img src="https://storage.googleapis.com/sourcegraph-assets/handbook/growth-and-integrations/join-us-vermillion.png" alt="Picture with CTA: Join Us" style="background: transparent; width:128px;"/>
        </div>
        <div style="text-align: center;">Stephen Gutekanst </div>
        <div style="text-align: center; font-size: 12px;">Software Engineer (growth)</div>
      </div>
    </div>
  </div>

</section>

## Growth vision

> 
> Sourcegraph is optimized to help users discover how powerful the tool is, as quickly as possible. This optimization leads to introductions to new users and customers. It also increases usage at customer accounts and with individual developers. By addressing these issues we will:
> 
> - Generate product qualified leads for the business
> - Expand usage in existing accounts
> - Increase the frequency of all users from MAU to DAU
> - Imrprove retention of all users

Author: ```suggestion
_Sourcegraph is optimized to help users discover how powerful the tool is, as quickly as possible._

This vision is necessarily broad because the growth-oriented goal for the team is the generation of Product Qualified Leads. These PQLs will generally be generated from users who find value in the product and integrate Sourcegraph into their daily routines. As such, this team will also impact the following objectives:

- Improve the retention of all users
- Increase the frequency of users from monthly active to daily active
- Expand usage in existing accounts
```

> - Expand usage in existing accounts

Author: @bcreager @christinaforney @nicksnyder Can you please have a look at this section explicitly? It lays out a necessarily broad vision for the growth team that attempts to credit and capture the advances we'll be making as we seek to generate PQLs.

Note that this vision statement doesn't equate to additional goal setting this quarter but sets us up for a longer-term vision and goal setting that will help us report on the individual _stages_ of the PQL generation.

Reviewer: I suggested an update to the vision statement to make it more concrete. The subsequent paragraphs feel like implementation details of the vision, so maybe we should label that differently.

I don't feel strongly about this, just suggestions.

Author: Thanks Nick. I like where we ended up with your help.

# Growth and Integrations team

> The growth and integrations team owns the initial user journey, growth-oriented projects, our code host and third-party integrations (including our browser extension), and Sourcegraph extensions.

Reviewer 2: Should we mention IDE extensions too?

Author: Good call. I've included them!

> The growth and integrations team owns the initial user journey, growth-oriented projects, our code host and third-party integrations (including our browser extension), and Sourcegraph extensions.

Author: ```suggestion
The growth and integrations team owns the initial user journey, growth-oriented projects, our code host and third-party integrations (including our IDE and browser extensions), and Sourcegraph extensions.
```

        <div style="text-align: center; font-size: 12px;">Software Engineer</div>
        <div style="text-align: center; font-size: 12px;">(starting 01/10)</div>
      </div>
    </div>

<div class="col" style="flex: 1;">
      <div>
        <div>
          <img src="https://storage.googleapis.com/sourcegraph-assets/handbook/growth-and-integrations/rob-rhyne.png" alt="Picture of Rob Rhyne" style="background: transparent; width:128px;"/>
        </div>
        <div style="text-align: center;">Rob Rhyne</div>
        <div style="text-align: center; font-size: 12px;">Temporary product manager and product designer, growth</div>
      </div>
    </div>
    <div class="col" style="flex: 1;">
      <div>
        <div>
          <img src="https://storage.googleapis.com/sourcegraph-assets/handbook/growth-and-integrations/join-us-vermillion.png" alt="Picture with CTA: Join Us" style="background: transparent; width:128px;"/>
        </div>
        <div style="text-align: center;">Stephen Gutekanst </div>
        <div style="text-align: center; font-size: 12px;">Software Engineer (growth)</div>
      </div>
    </div>
  </div>

</section>

## Growth vision

> _Sourcegraph is optimized to help users discover how powerful the tool is, as quickly as possible._

Reviewer: What do you think about something more like this? (more concrete, IMO)

```suggestion
Every developer who visits a Sourcegraph website becomes a regular user who invites other people they know to use Sourcegraph.
```

Author: I don't think that's a reasonably attainable goal for this team. Strictly read, it would require additions to the product to bring value to developer segments we currently do not serve.

Reviewer: > it would require additions to the product to bring value to developer segments we currently do not serve.

Isn't our vision for Sourcegraph to be a daily use case for every developer?

I think the question here is time horizon. If we are talking about this year, I completely agree with you, but if we are talking about an ambitious vision that will fuel a team in perpetuity, then is that bad?

> _Sourcegraph is optimized to help users discover how powerful the tool is, as quickly as possible._

Author: ```suggestion
_Visitors to Sourcegraph clearly see the value of the product and are compelled to integrate it into their workflow and share it with friends and colleagues. _
```

@nicksnyder , I felt my original statement was not strong as well, so I tried to rephrase a bit with the help of what I think is part of your suggestion's intent.

Reviewer: lgtm!

Reviewer 3: looks great! Solid initial overarching vision while we discover the right product metric to optimize

This vision is necessarily broad because the growth-oriented goal for the team is the generation of Product Qualified Leads. These PQLs will generally be generated from users who find value in the product and integrate Sourcegraph into their daily routines. As such, this team will also impact the following objectives:

- Improve the retention of all users
- Increase the frequency of users from monthly active to daily active

> - Expand usage in existing accounts

Reviewer 2: Are we going to own the "Acquisition" and "Activation" steps, too, such as from organic search, landing pages, possible sign-up wizards (similar to onboarding tour) etc.?

Author: Great question. Acquisition we will not own. We must own parts of activation as it is a critical step on the path to PQL generation. 

We may also work to increase the acquisition of specific key moments in the product, such as improving the install rate of the browser extensions and IDE extensions. We would experiment in these areas where it helps us with our end goal of improving PQLs.

> - Expand usage in existing accounts

Author: @bcreager @christinaforney pinging you again on this. Can you please have a look at this section explicitly? It lays out a necessarily broad vision for the growth team that attempts to credit and capture the advances we'll be making as we seek to generate PQLs. Said more plainly, to generate a PQL we need active and retained users, so growth should be working in those areas.

Note that this vision statement doesn't equate to additional goal setting this quarter but sets us up to properly measure the stages of the user journey we'll need to improve to generate PQLs. 

cc @anna-mmikhova

Reviewer 3: @rrhyne I'm good with this to start. Signals a focus from the team on Activation and Retention of new and existing users which in turn should drive PQLs and account expansion, and serving both Cloud and On-Prem deployment.

## Integrations vision

> Sourcegraph powers your first stop for any information about your code, for any tool or website you use.

Reviewer: What do you think about something more like this:

```suggestion
The power of Sourcegraph is available everywhere that developers read or write code.
```

- Build a healthy ecosystem of Sourcegraph extensions, built by developers everywhere, who want to bring the power of existing developer tools they love into our platform.
- Build new paths to get value from Sourcegraph on other websites and IDEs.

> - All of Sourcegraph's most powerful features are easily available on most any code host while using any browser thanks to native integrations and browser extensions.

Reviewer: This section feels like it contains some implementation details of how we achieve the vision. As such, perhaps this belongs in in a section that describes the strategy of how the team is thinking about achieving the vision.

Author: Nick, there are no changes here from the previous iteration. If you feel strongly about this, perhaps @muratsu could address?

Reviewer: All my comments here are non-blocking (I approved the PR), so your discretion on whether you want to update this.

## Responsibilities and focus

### Growth-oriented responsibilities

In pursuit of goals related to growth, the team will own or contribute to the following:

- Researching how users become activated, sign-up, and generate PQLs
- Guiding and shaping the initial user journey of:
  - visitors to sourcegraph.com
  - authenticated users of Self-hosted installations
  - TBD: authenticated users to the cloud product
    - Cloud is working on this, but solutions here probably need to be influenced by user journeys developed for the self-hosted product and vice-versa
- Creation and optimization of in-product flows that result in:
  - Cloud sign-ups

>   - Self-hosted installs

Reviewer 2: To clarify, we are in Cloud org, but growth efforts are oriented in both Cloud and Self-hosted?

Author: Yes, and this is where it gets confusing for lots of folks. A PQL is for either a Cloud account or a Self-hosted account install. 

At the part of the user journey we are working to improve, we cannot determine which product a given user needs. We can only help them see value and want to bring that value to their team. We will then guide them to a place where they choose the right product for them. 

Because we worked to get the user to either of those actions, it makes sense that we also acknowledge that we will create these events.

- [team/growth-and-integrations](https://github.com/sourcegraph/sourcegraph/labels/team%2Fgrowth-and-integrations) label and @sourcegraph/growth-and-integrations team on GitHub.

## Additional resources

- [Processes](processes.md)
- [Goals](../../../../company/strategy/cloud/growth-and-integrations/index.md)
- Products
  - [Browser Extensions](browser-extensions/index.md)
  - [Sourcegraph Extensions](https://docs.sourcegraph.com/extensions)
  - [IDE Extensions](ide-extensions/index.md)
- [Role of sourcegraph.com/search in revenue generation](sourcegraph.com-and-revenue.md)
- [Defining ‘growth team’](what-is-a-growth-team.md)
- [Team history](team-history.md)

## Tech stack

We use a modern, flexible tech stack. Here are some of the technologies we use to deliver on our goals:

- [TypeScript](https://www.typescriptlang.org/)
- [React](https://reactjs.org/)
- [SCSS](https://sass-lang.com/) + [Bootstrap](https://getbootstrap.com/)
- [GraphQL](https://graphql.org/)
- [WebExtensions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API)
- [RxJS](https://rxjs-dev.firebaseapp.com/guide/overview)

> - [Web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API)

Reviewer 2: We sometimes make changes to the backend (Golang) and database (Postgres), although not often.

Author: Feel free to add a line here!

In March 2016, [Sourcegraph released its first browser extension](https://about.sourcegraph.com/blog/browse-review-code-on-github-like-in-an-ide-with-the-sourcegraph-chrome-extension/) so that you can take the value of Sourcegraph to your code host for code reviews.

In October 2018, [Sourcegraph released the extensions API](https://about.sourcegraph.com/blog/sourcegraph-2-12-release-notes/) so that others could build on top of Sourcegraph, and so that Sourcegraph could integrate with your other tools.

In June 2019, [Sourcegraph created a BitBucket native integration](https://github.com/sourcegraph/bitbucket-server-plugin/commit/e450abf50c128fa5ee18439ff93e0631e4868de7) so you don't need to install the browser extension to get the value of Sourcegraph on your BitBucket code host.

In November 2019, [Sourcegraph partnered with GitLab to create a native integration](https://about.gitlab.com/blog/2019/11/12/sourcegraph-code-intelligence-integration-for-gitlab/).

In December 2020, [Sourcegraph released a Safari Extension](https://apps.apple.com/us/app/sourcegraph-for-safari/id1543262193) to support Safari users with our browser extension, in addition to Chrome and Firefox.

> In early 2021, the Growth and Integrations team was formed as an offshoot of the "web team" (which became the 2021 Developer Insights org) because the product direction, support and maintenance, validation, and future promises of Sourcegraph's browser extensions, native code host integrations, and Sourcegraph extensions required and deserved the resources of its own team.
> 
> In late 2021, the Growth and Integrations team was renamed Growth and Integrations.

Reviewer 2: ```suggestion
In early 2021, the Extensibility team was formed as an offshoot of the "web team" (which became the 2021 Developer Insights org) because the product direction, support and maintenance, validation, and future promises of Sourcegraph's browser extensions, native code host integrations, and Sourcegraph extensions required and deserved the resources of its own team.

In late 2021, the Extensibility team was renamed Growth and Integrations.
```
